src/containers/SignIn/index.tsx

```diff
@@ -20,10 +20,13 @@ const SignIn: FC = () => {
   const dispatch = useDispatch<AppDispatch>();
   const history = useHistory();
   const [errorMessage, setErrorMessage] = useState<string>('');
+  const [submitting, setSubmitting] = useState<boolean>(false);
 
   const handleSubmit = async ({email, password}: FormValues): Promise<void> => {
     try {
+      setSubmitting(true);
       const {data} = await dispatch(login({email, password}));
+      setSubmitting(false);
       history.push(`/users/${data.user.pk}`);
     } catch (error) {
       setErrorMessage(formatAPIError(error));
```

Comment:
you can remove the `setSubmitting(false)` from line 29, and instead, below the `catch` block, put a `finally { setSubmitting(false); }`

Reply:
Don't think putting in the `finally` block is the best as the `setSubmitting(false)` will be after `history.push(...)`. That will lead to a unable to setState after component is unmounted error.

I added another `setSubmitting(false)` to the `catch` block instead

Reply:
yeah good point